### ISSUE TITLE
Potential bug in target.py

### ISSUE THREAD

**Issue by khvitri** [ucsb-seclab/crush#1]
Hi! I'm wondering if the following code snippet in `def _format_calldata(...)` is a potential bug:

```python
if not callsite_state.solver.is_formula_sat(BV_UGE(calldata_length, length)):
            # be dumb, skim off a few bytes and retry
            calldata = calldata[:-32]
            calldata_length = BVV(len(calldata) // 2, 256)
            while not callsite_state.solver.is_formula_sat(BV_UGE(calldata_length, length)):
                calldata = calldata[:-32]
                calldata_length = BVV(len(calldata) // 2, 256)

if calldata_length.value == 0:
            log.error("Could not fit calldata in memory buffer")
            return raw_calldata
```

From my understanding, you are trying to fit `calldata_length` (the calldata you are trying to reformat) into `length` (the memory buffer of `callsite_state` to store arguments for the `DELEGATECALL` opcode). However, the code snippet above doesn't do this because it's only reducing `calldata_length` when `calldata_length` < `length`. So instead shouldn't the code look more like:

```python
if callsite_state.solver.is_formula_sat(BV_UGE(calldata_length, length)):
            # be dumb, skim off a few bytes and retry
            calldata = calldata[:-32]
            calldata_length = BVV(len(calldata) // 2, 256)
            while callsite_state.solver.is_formula_sat(BV_UGE(calldata_length, length)):
                calldata = calldata[:-32]
                calldata_length = BVV(len(calldata) // 2, 256)

if calldata_length.value == 0:
            log.error("Could not fit calldata in memory buffer")
            return raw_calldata
```

This time I just removed the "not". Please correct me if I'm wrong though. I'm still familiarizing myself with this project! 😊

**Comment by ruaronicola** [comment]
Yes, this looks like a bug. The two operands of BV_UGE should be swapped or else the operator should be BV_ULE. I just pushed a fix in the latest commit (#f1b2f07b5a49f744865524c3e7af52add0feae62). Thanks for reporting!

**Comment by ruaronicola** [comment]
Yes, this looks like a bug. The two operands of BV_UGE should be swapped or else the operator should be BV_ULE. I just pushed a fix in the latest commit. Thanks for reporting! :)
